pkg/services/folder/folderimpl/unifiedstore.go

```diff
@@ -0,0 +1,467 @@
+	if !ok {
+		return nil, nil
+	}
+
+	obj, err := internalfolders.LegacyCreateCommandToUnstructured(&cmd)
+	if err != nil {
+		return nil, err
+	}
+	out, err := client.Create(newCtx, obj, v1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	folder, _ := internalfolders.UnstructuredToLegacyFolder(*out, cmd.SignedInUser.GetOrgID())
+	if err != nil {
+		return nil, err
+	}
+
+	return folder, err
+}
+
+func (ss *FolderUnifiedStoreImpl) Delete(ctx context.Context, UIDs []string, orgID int64) error {
+	newCtx, cancel, err := ss.getK8sContext(ctx)
+	if err != nil {
+		return err
+	} else if cancel != nil {
+		defer cancel()
+	}
+
+	client, ok := ss.k8sclient.getClient(newCtx, orgID)
+	if !ok {
+		return nil
+	}
+
+	for _, uid := range UIDs {
+		err = client.Delete(newCtx, uid, v1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (ss *FolderUnifiedStoreImpl) Update(ctx context.Context, cmd folder.UpdateFolderCommand) (*folder.Folder, error) {
+	newCtx, cancel, err := ss.getK8sContext(ctx)
+	if err != nil {
+		return nil, err
+	} else if cancel != nil {
+		defer cancel()
+	}
+
+	client, ok := ss.k8sclient.getClient(newCtx, cmd.OrgID)
+	if !ok {
+		return nil, nil
+	}
+
+	obj, err := client.Get(ctx, cmd.UID, v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	updated, err := internalfolders.LegacyUpdateCommandToUnstructured(obj, &cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	out, err := client.Update(ctx, updated, v1.UpdateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	folder, _ := internalfolders.UnstructuredToLegacyFolder(*out, cmd.SignedInUser.GetOrgID())
+	if err != nil {
+		return nil, err
+	}
+
+	return folder, err
+}
+
+// If WithFullpath is true it computes also the full path of a folder.
+// The full path is a string that contains the titles of all parent folders separated by a slash.
+// For example, if the folder structure is:
+//
+//	A
+//	└── B
+//	    └── C
+//
+// The full path of C is "A/B/C".
+// The full path of B is "A/B".
+// The full path of A is "A".
+// If a folder contains a slash in its title, it is escaped with a backslash.
+// For example, if the folder structure is:
+//
+//	A
+//	└── B/C
+//
+// The full path of C is "A/B\/C".
+func (ss *FolderUnifiedStoreImpl) Get(ctx context.Context, q folder.GetFolderQuery) (*folder.Folder, error) {
+	// create a new context - prevents issues when the request stems from the k8s api itself
+	// otherwise the context goes through the handlers twice and causes issues
+	newCtx, cancel, err := ss.getK8sContext(ctx)
+	if err != nil {
+		return nil, err
+	} else if cancel != nil {
+		defer cancel()
+	}
+
+	client, ok := ss.k8sclient.getClient(newCtx, q.OrgID)
+	if !ok {
+		return nil, nil
+	}
+
+	out, err := client.Get(newCtx, *q.UID, v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	dashFolder, _ := internalfolders.UnstructuredToLegacyFolder(*out, q.SignedInUser.GetOrgID())
+	return dashFolder, nil
```

Comment:
```suggestion
	return internalfolders.UnstructuredToLegacyFolder(*out, q.SignedInUser.GetOrgID())
```

Reply:
ouch!! I had to revert this. The second parameter returned by `UnstructuredToLegacyFolder` is not an error, it's a string containing the `createdBy` user.